src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -115,6 +115,7 @@ public void PathAlreadyExistsAsDirectory(FileAttributes attributes)
         }
 
         [Fact]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotInAppContainer))] // Can't read root in appcontainer        
```

Comment:
Do we need a test validating what the behavior actually is in an app container?

Reply:
Generally I don't think so since other tests would fail if error handling did not go through GetExceptionForWin32Error, which we know handles ERROR_ACCESS_DENIED; also we are not trying to check the appcontainer sandboxing.

However CreateDirectory() does have special handling for ERROR_ACCESS_DENIED so I added a test.